integration/app_integration_test.go

```diff
@@ -64,12 +64,74 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// TestAppAccessForward tests that requests get forwarded to the target application
-// within a single cluster and trusted cluster.
-func TestAppAccessForward(t *testing.T) {
-	// Create cluster, user, sessions, and credentials package.
+// TestAppAccess runs the full application access integration test suite.
+//
+// It allows to make the entire cluster set up once, instead of per test,
+// which speeds things up significantly.
+func TestAppAccess(t *testing.T) {
 	pack := setup(t)
 
+	t.Run("TestAppAccessForward", func(t *testing.T) {
+		AppAccessForward(t, pack)
+	})
```

Comment:
nit: If you changed these tests to be a method on pack this might read a bit cleaner

```
func (p *pack) appAccessForward(t *testing.T) { ... }

t.Run("TestAppAccessForward", p.appAccessForward)
```

Reply:
I think another good option is testify suite https://pkg.go.dev/github.com/stretchr/testify@v1.7.5/suite